generic3g/specs/FieldSpec.F90

```diff
@@ -916,33 +911,64 @@ logical function adapter_match_vertical_grid(this, spec) result(match)
 
    contains
 
-     logical function same_vertical_grid(src_grid, dst_grid)
-        class(VerticalGrid), intent(in) :: src_grid
-        class(VerticalGrid), allocatable, intent(in) :: dst_grid
-         
+      logical function same_vertical_grid(src_grid, dst_grid)
+         class(VerticalGrid), intent(in) :: src_grid
+         class(VerticalGrid), allocatable, intent(in) :: dst_grid
+
          same_vertical_grid = .true.
          if (.not. allocated(dst_grid)) return ! mirror grid
-         
+
          same_vertical_grid = src_grid%same_id(dst_grid)
 
          block
-           use mapl3g_BasicVerticalGrid
-           ! "temporary kludge" while true vertical grid logic is being implemented
-           if (.not. same_vertical_grid) then
-              select type(src_grid)
-              type is (BasicVerticalGrid)
-                 select type (dst_grid)
-                 type is (BasicVerticalGrid)
-                    same_vertical_grid = (src_grid%get_num_levels() == dst_grid%get_num_levels())
-                 end select
-              end select
-           end if
+            use mapl3g_BasicVerticalGrid
```

Comment:
Let's try to eliminate the BLOCK in this PR.   I know it pre-existed, but ... BLOCK is bad for anything other than a temporary kludge.  The time has come ...

Reply:
Yes! But not in this PR. I'll soon have another one where we regrid VerticalGrid. It'll be part of that.